Sources/CacheAdvance/CacheAdvance.swift

```diff
@@ -55,26 +53,36 @@ public final class CacheAdvance<T: Codable> {
 
     public let fileURL: URL
 
+    /// Checks if the all the header metadata provided at initialization matches the persisted header. If not, the cache is not writable.
+    /// - Returns: `true` if the cache is writable.
+    public func isWritable() throws -> Bool {
+        try setUpFileHandlesIfNecessary()
+        return try header.canWriteToFile()
+    }
+
     /// Appends a message to the cache.
     /// - Parameter message: A message to write to disk. Must be smaller than both `maximumBytes - FileHeader.expectedEndOfHeaderInFile` and `MessageSpan.max`.
     public func append(message: T) throws {
         try setUpFileHandlesIfNecessary()
+        guard try header.canWriteToFile() else {
+            throw CacheAdvanceError.fileNotWritable
+        }
 
         let encodableMessage = EncodableMessage(message: message, encoder: encoder)
         let messageData = try encodableMessage.encodedData()
         let bytesNeededToStoreMessage = Bytes(messageData.count)
 
         guard
-            bytesNeededToStoreMessage <= maximumBytes - FileHeader.expectedEndOfHeaderInFile // Make sure we have room in this file for this message.
+            bytesNeededToStoreMessage <= header.maximumBytes - FileHeader.expectedEndOfHeaderInFile // Make sure we have room in this file for this message.
```

Comment:
Good point. I was also thinking of renaming `CacheHeaderHandle` -> `CacheHeader`, since it's more than just a thin wrapper on a handle now. Will tackle both of these thoughts at the same time.

Reply:
Yeah I like the idea of hiding the "handle" concept as much as possible. Hard to think about.